### DISCUSSION TITLE
How do I test my changes to multiplayer?

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#16993] [Q&A] [answered]
I would like to make some changes to multiplayer rooms in osu! but when I run `dotnet run --project osu.Desktop` and go to multiplayer, it says I gotta be online, which I am. If I try to login, it doesn't work either, even with correct login info.

I think the issue is that I'm not connected to the database or something because I'm running my own local build.

I'm probably missing something obvious but isn't there any easy way to get to something like a fake multiplayer room just so I can see and test my changes?

In case it makes a difference, I want to try making a PR that adds the `PlaylistRemoveButton` to the first item in a playlist in multiplayer rooms as well:
![image](https://user-images.githubusercontent.com/35064754/155843578-7d01fec2-29e4-4d10-8f3e-be6dd1d7d31f.png)
Currently it's not there and it's very annoying if you just want to skip that song and remove it. Instead you have to edit which is rather inconvenient. But if there is a discussion about this anywhere saying that this should be kept the way it is, I might as well abandon my efforts trying to do this change.

**Comment by bdach** [comment, marked as answer]
Let me preempt the actual answer by this: adding removal of current playlist item requires server-side considerations [and there are known limitations that make it not so easy](https://github.com/ppy/osu-server-spectator/pull/93#discussion_r765734184). Probably need to double-check that @smoogipoo doesn't have any changes for this before proceeding. In general it is advised to **not** PR changes unadvised unless there is already an issue open for them with a clear provided direction.

As for your question: In debug configuration the lazer client uses a development instance of the website at `dev.ppy.sh`. You need to register an account there to continue. However making this change would require setting up: lazer, the website and the spectator server locally. Instructions for this are provided [in this wiki article](https://github.com/ppy/osu/wiki/Testing-web---server-full-stack-with-osu!).

**Reply by ghost** [reply]
> Let me preempt the actual answer by this: [...]

Oh, I understand now. It does seem a bit tricky to do this change then. Seems like it's better to leave it as-is for now.

> As for your question: [...]

Ah yes, the wiki is what I was missing. I knew there had to be a place with more information about development of osu! itself. Good to know for next time.

Thank you very much.
